test/suites/dev-tanssi/registrar/test_registrar_proxy.ts

```diff
@@ -0,0 +1,107 @@
+                        delay: 0,
+                    },
+                ]);
+
+                // Use proxy
+                await context.createBlock();
+                const currentSesssion = await polkadotJs.query.session.currentIndex();
+                const sessionDelay = await polkadotJs.consts.registrar.sessionDelay;
+                const expectedScheduledOnboarding =
+                    BigInt(currentSesssion.toString()) + BigInt(sessionDelay.toString());
+
+                const emptyGenesisData = () => {
+                    const g = polkadotJs.createType("TpContainerChainGenesisDataContainerChainGenesisData", {
+                        storage: [
+                            {
+                                key: "0x636f6465",
+                                value: "0x010203040506",
+                            },
+                        ],
+                        name: "0x436f6e7461696e657220436861696e2032303030",
+                        id: "0x636f6e7461696e65722d636861696e2d32303030",
+                        forkId: null,
+                        extensions: "0x",
+                        properties: {
+                            tokenMetadata: {
+                                tokenSymbol: "0x61626364",
+                                ss58Format: 42,
+                                tokenDecimals: 12,
+                            },
+                            isEthereum: false,
+                        },
+                    });
+                    return g;
+                };
+                const containerChainGenesisData = emptyGenesisData();
+
+                // assert we can inject on chain data with proxy
+                const tx2 = polkadotJs.tx.proxy.proxy(
+                    alice.address,
+                    null,
+                    polkadotJs.tx.registrar.register(2002, containerChainGenesisData)
+                );
+                await context.createBlock([await tx2.signAsync(charlie)]);
+                // Check that the on chain genesis data is set correctly
+                const onChainGenesisData = await polkadotJs.query.registrar.paraGenesisData(2002);
+                // TODO: fix once we have types
+                expect(emptyGenesisData().toJSON()).to.deep.equal(onChainGenesisData.toJSON());
+
+                // assert we can inject bootnodes with proxy
+                const tx3 = polkadotJs.tx.proxy.proxy(
+                    alice.address,
+                    null,
+                    polkadotJs.tx.registrar.setBootNodes(2002, ["dummy"])
+                );
+                await context.createBlock([await tx3.signAsync(charlie)]);
+
+                // Check that the on chain genesis data is set correctly
+                const onChainBootnodes = await polkadotJs.query.registrar.bootNodes(2002);
+                // TODO: fix once we have types
+                expect(onChainBootnodes.toHuman()).to.deep.equal(["dummy"]);
+            },
```

Comment:
This test could also check that the registrar deposit is set to alice when charlie registers the chain using proxy. And also add a test for sudoRegistrar, it should be easy, try calling sudo(markValidForCollating) through the proxy.